src/components/collapsible-panel.tsx

```diff
@@ -0,0 +1,43 @@
+import { ToolbarButtonComponent } from '@jupyterlab/apputils';
+import React, { useState } from 'react';
+import { caretDownIcon, caretRightIcon } from '@jupyterlab/ui-components';
+
+export interface ICollapsiblePanelProps {
+  expandedByDefault?: boolean;
+  title: JSX.Element | string;
+  content: JSX.Element | string;
+}
+
+export function CollapsiblePanel(props: ICollapsiblePanelProps): JSX.Element {
+  const [expanded, setExpanded] = useState(props.expandedByDefault ?? false);
+
+  const onControlClick = () => {
+    setExpanded(!expanded);
+  };
+
+  return (
+    <div className={'jp-collapsible-panel' + (expanded ? ' expanded' : '')}>
+      <div
+        className={
+          'jp-collapsible-panel-summary' + (expanded ? ' expanded' : '')
+        }
+      >
+        <ToolbarButtonComponent
+          className="jp-collapsible-panel-icon-toolbar-button"
+          icon={expanded ? caretDownIcon : caretRightIcon}
+          onClick={onControlClick}
+        />
```

Comment:
I originally used `Button` for the action icons in each row, but even the "minimal" styling on buttons drove me away from `Button` and towards `ToolbarButtonComponent`. Displaying the icon inline could work as an alternative, though, since we don't intend to use it as a button by itself.

Reply:
Got rid of `ToolbarButtonComponent` by using `caretDownIcon/caretRightIcon.react` to render the icon inline directly